### PR TITLE
Add support for `avg`/`average` aggregate keywords

### PR DESCRIPTION
**Pull request by vishaltelangre**
### What

Similar to https://github.com/vishaltelangre/NerdCalci/pull/17.

Adds `avg` and `average` as aggregate keywords that automatically compute the average of all line results in the current block (up to the nearest blank/comment/error line or start of file).

- `avg` and `average` are aliases, i.e. both evaluate to the same block average
- Can be used standalone (`average`) or inline in expressions (`average * 0.5`)
- Blank/comment/error lines act as block boundaries, resetting the aggregate
- Explicit reassignment (`avg = 40`) overrides the aggregate meaning from that point on

#### Examples:

Use `avg` or `average` to get the average of all line results above, up to the nearest blank/comment/error line.

```text
feb = 200
mar = 300
avg                 # evaluates to 250
```

Blank lines create blocks — `avg`/`average` only averages within the current block:

```text
a = 10
b = 20
average             # evaluates to 15

c = 5
avg                 # evaluates to 5
```

Use `avg` or `average` inside expressions:

```text
item1 = 25
item2 = 75
half_avg = avg / 2  # evaluates to 25
```

Assigning to `avg` or `average` overrides the aggregate meaning from that point onward:

```text
a = 10
b = 20
avg                 # evaluates to 15

avg = 40
avg / 2             # evaluates to 20 (uses assigned value, not aggregate)
avg                 # evaluates to 40
# i.e. after assigning, avg no longer aggregates
```

### Why

Users often want a running average without manually calculating the sum and dividing by the number of items.

## Summary by CodeRabbit

* **New Features**
  * Added `avg` / `average` aggregate keywords to compute block averages, usable inline and overridable by assignment.

* **Documentation**
  * Expanded docs and app store description with Aggregates section, examples, and clarified behavior across blank/comment/error boundaries; updated app tagline.

* **Tests**
  * Added comprehensive tests covering avg/average behavior, edge cases, block interactions, and assignment semantics.

* **Chores**
  * Bumped app version to 1.9.1.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,7 +28,7 @@
 ### Smart Calculations
 
 > [!NOTE]
-> For a complete language reference of all operators, functions, and advanced syntax, please see the [Language Reference Guide](REFERENCE.md).
+> For a complete reference of all supported operators, functions, and advanced syntax is documented in the [reference guide](REFERENCE.md).
```
[the remark this answers is not in the document]
Comment (vishaltelangre) [reply README.md:31]: Fixed.
